### PR TITLE
INTYGFV-15088: Implementation of new component QuestionSynskarpa with tests. Added new CertificateDataConfig UE_VISUAL_ACUITY. Added new CertificateDataValue VISUAL_ACUITIES, VISUAL_ACUITY & DOUBLE.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/util/SynskarpaEnum.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.common.ts_bas.v7.model.converter.util;
+
+public enum SynskarpaEnum {
```
Comment (mww-sopra-steria) [inline ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/util/SynskarpaEnum.java:22]: Maybe this enum should be defined together with the QuestionSynskarpa. It is only used in order to specify what value to retrieve when calling toInternal.

File: support/src/main/java/se/inera/intyg/common/support/facade/model/config/VisualAcuity.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import lombok.Builder;
+import lombok.Value;
+import se.inera.intyg.common.support.facade.model.config.VisualAcuity.VisualAcuityBuilder;
+
+@JsonDeserialize(builder = VisualAcuityBuilder.class)
+@Value
+@Builder
+public class VisualAcuity {
```
Comment (mww-sopra-steria) [inline support/src/main/java/se/inera/intyg/common/support/facade/model/config/VisualAcuity.java:31]: There are two things to consider in the config:
1) What precision is the visualAcuity value i.e. how many decimals.
2) What are the labels that represent contact lenses true/false i.e. in this case Ja/Nej.

Both can be handled in the configuration or they could be hardcoded on the client. As this is a custom tool for VisualAcuity, I think it is OK to let it be hardcoded. What are your thoughts?

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
I have to say, really awesome work! I'm really happy to review this PR! :-)

Have a look at the comments and see if we should make some modifications before it is ready to be merged. But except those minor comments it is pretty much perfect in my view! :-)

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionSynskarpa.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        return VisualAcuity.builder()
+            .label(textProvider.get(HOGER_OGA_LABEL_ID))
+            .withoutCorrectionId(HOGER_OGA_UTAN_KORREKTION_DELSVAR_ID_8)
+            .withCorrectionId(HOGER_OGA_MED_KORREKTION_DELSVAR_ID_8)
+            .contactLensesId(KONTAKTLINSER_HOGER_OGA_DELSVAR_ID_8)
+            .build();
+    }
+
+    public static Synskarpevarden toInternal(Certificate certificate, SynskarpaEnum type) {
+        final var value = (CertificateDataValueVisualAcuities) certificate.getData().get(VARDEN_FOR_SYNSKARPA_ID).getValue();
```
Comment (mww-sopra-steria) [inline ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionSynskarpa.java:199]: Return empty Synskarpevarden if getData().get returns null

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/util/SynskarpaEnum.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/util/SynskarpaEnum.java:22]: Refactored into a inner enum in QuestionSynskarpa and renamed to VisualAcuityEnum

File: support/src/main/java/se/inera/intyg/common/support/facade/model/config/VisualAcuity.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply support/src/main/java/se/inera/intyg/common/support/facade/model/config/VisualAcuity.java:31]: After discussion we decided to keep the original implementation

File: ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionSynskarpa.java (hunk 3 above)
Comment (ChrilleSigerhed) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionSynskarpa.java:199]: Check!

Comment (mww-sopra-steria) [reply ts/ts-bas/src/main/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionSynskarpa.java:198]: Sorry for not being completely clear in my comment... My comment was to add a null check on certificate.getData().get(VARDEN_FOR_SYNSKARPA_ID) 

But maybe it is best to make sure do check null for getValue() as well

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Sorry if my comment before was misleading... please take a look at the new comment.

**Pull request merged**